test/unit/job_invocation_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_plugin_helper'
+
+describe JobInvocation do
+
+  let(:job_invocation) { FactoryGirl.build(:job_invocation) }
+  let(:template) { FactoryGirl.build(:job_template) }
+
+  context 'Able to be created' do
+    it { assert job_invocation.save }
+  end
+
+  context 'Requires targeting' do
+    before { job_invocation.targeting = nil }
+
+    it { refute job_invocation.save }
+  end
+
+  context 'has template invocations' do
+    before do
+      job_invocation.template_invocations << TemplateInvocation.new(:template => template)
+      job_invocation.save!
+    end
+
+    it { refute job_invocation.reload.template_invocations.empty? }
+  end
+
+  context 'has template input values' do
+    let(:template) { FactoryGirl.create(:job_template, :with_input) }
+    let(:input_value) { FactoryGirl.build(:template_invocation_input_value) }
+
+    before do
+      job_invocation.template_invocations << TemplateInvocation.new(:template => template)
+      job_invocation.save!
```

Comment:
these two lines seems same as in previous context, I think it would be better to nest it in it and just define input values (or if you disagree, use FactoryGirl for building invocation)